core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java

```diff
@@ -283,6 +289,20 @@ public TupleFilter acceptOptimizeTransformer(FilterOptimizeTransformer transform
         return transformer.visit(this);
     }
 
+    private TupleFilter optimizeChildCompareTupleFilter(CompareTupleFilter child) {
+        FilterOptimizeTransformer transformer = new FilterOptimizeTransformer();
+        TupleFilter result = child.acceptOptimizeTransformer(transformer);
+        if (result == ConstantTupleFilter.TRUE) {
+            // use string instead of boolean since it's encoded as string
+            result = new ConstantTupleFilter("true");
+        } else if (result == ConstantTupleFilter.FALSE) {
+            result = new ConstantTupleFilter("false");
+        } else {
+            this.isNormal = false;
```

Comment:
this means that TupleFilters excepted AlwaysTrue or AlwaysFalse will not pushdown?

Reply:
Why do you think AlwaysTrue or AlwaysFalse cannot be pushed down? Current CompareTupleFilter does not well support the case that both children are CompareTupleFilter. Therefore, push down will be disabled. This behavior is not changed.